stylefunction.js

```diff
@@ -601,6 +601,19 @@ export default function(olLayer, glStyle, source, resolutions, spriteData, sprit
             label = label.toLowerCase();
           }
           const wrappedLabel = type == 2 ? label : wrapText(label, font, getValue(layer, 'layout', 'text-max-width', zoom, f));
+          const letterSpacing = getValue(layer, 'layout', 'text-letter-spacing', zoom, f);
+          if (letterSpacing && letterSpacing > 0.01 /* default is 0.01 */ ) {
+            let wrappedLabelWithLetterSpacing = '';
+            wrappedLabel.split('\n').forEach((line) => {
+              let joinSpaceString = "",
+                  spacingCount;
+              for (spacingCount = 0; spacingCount < letterSpacing; spacingCount += 0.1) {
```

Comment:
No need to do this in a loop. The `joinSpaceString` can be created with
```js
const joinSpaceString = Array(Math.round(letterSpacing / 0.1)).join('\u200A');
```
This way we add one hair space for a `letterSpacing` of `0.5` to `1.499999`, two for `1.5` to `2.499999` and so on.

Reply:
Isnt't join also part of the filter map reduce, and thus will be a bigger impact as to do a simple for loop?

Reply:
You're right regarding performance. And both your and my approach are lame. I've come up with a 3rd way, see https://jsperf.com/pad-strings. So the best is to define a string pool only once that we slice from:
```js
// on the module level
const hairSpacePool = Array(256).join('\u200A');

export function styleFunction(feature, resolution) {
  // ...
  const joinSpaceString = hairSpacePool.slice(0, Math.round(letterSpacing / 0.1));
  // ...
}